### PR TITLE
support configurable parents and fix EVP issues between OSSL versions

### PR DESCRIPTION
**Pull request by williamcroberts**
When closed, make sure to twiddle task list items in: https://github.com/tpm2-software/tpm2-tools/issues/1079

### CODE REVIEW

File: tools/tpm2_import.c [hunk 1]
```diff
@@ -335,36 +340,80 @@ static void hmac_outer_integrity(uint8_t *buffer1, uint16_t buffer1_size,
...
+        0xBA, 0xDC, 0xC0, 0xDE, 0xBA, 0xDC, 0xC0, 0xDE,
+        0xBA, 0xDC, 0xC0, 0xDE, 0xBA, 0xDC, 0xC0, 0xDE,
+        0xBA, 0xDC, 0xC0, 0xDE, 0xBA, 0xDC, 0xC0, 0xDE,
+        0xBA, 0xDC, 0xC0, 0xDE, 0xBA, 0xDC, 0xC0, 0xDE,
+        0xBA, 0xDC, 0xC0, 0xDE, 0xBA, 0xDC, 0xC0, 0xDE,
+        0xBA, 0xDC, 0xC0, 0xDE, 0xBA, 0xDC, 0xC0, 0xDE,
+    };
+
+    ctx.protection_seed_data.size = tpm2_alg_util_get_hash_size(ctx.name_alg);
+    //RAND_bytes(ctx.protection_seed_data.buffer, TPM2_SHA256_DIGEST_SIZE); //max tpm digest
```
Comment (williamcroberts) [inline tools/tpm2_import.c]: FAIL!

File: man/tpm2_import.1.md [hunk 2]
```diff
@@ -64,6 +71,11 @@ tpm2_import -Q -G rsa -k private.pem -C 0x81010005 -f parent.pub \
 -q import_rsa_key.pub -r import_rsa_key.priv
 ```
 
+# LIMITATIONS
+  * The TPM requires that the name algorithm of the child be smaller then the parent.
```
Comment (williamcroberts) [inline man/tpm2_import.1.md]: smaller than the parent.

**Comment by williamcroberts** [comment]
@diabonas the second commit db4c126 should address your OSSL issue filed in #1118

**Comment by diabonas** [comment]
@williamcroberts Great, thank you very much! I can confirm that with this pull request everything compiles fine with OpenSSL 1.1.0h.

**Comment by williamcroberts** [comment]
Ahh scan build fail:
```
  CC       tools/tpm2_makecredential.o
../tools/tpm2_import.c:398:16: warning: Potential leak of memory pointed to by 'concatenated_seed_unique'
        return false;
               ^~~~~
/usr/lib/llvm-3.8/bin/../lib/clang/3.8.0/include/stdbool.h:33:15: note: expanded from macro 'false'
#define false 0
```

Ill fix it tomorrow evening.

File: tools/tpm2_import.c [hunk 3]
```diff
@@ -592,16 +645,20 @@ static void calculate_outer_integrity(void) {
 }
 
 static void create_import_key_private_data(void) {
-    ctx.import_key_private.size = sizeof(uint16_t) + TPM2_SHA256_DIGEST_SIZE
+
+    //UINT16 hash_size = tpm2_alg_util_get_hash_size(ctx.name_alg);
```
[the remark this answers is not in the document]
Comment (williamcroberts) [reply tools/tpm2_import.c]: In the hardcoded SHA256 maze, I wasn't sure if this should be parent or child name algorithm. Turned out to be parent and I left the line in.

**Pull request merged**